### PR TITLE
Fix doughnut graphs on workload overview sometimes not showing

### LINKED ISSUE #5706: Workloads overview is missing donut graphs if a namespace is empty
**Describe the bug**

image

**Environment (please complete the following information):**
- Lens Version: 5.6.0-alpha.3 (or master)

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #5706

### CODE REVIEW

File: src/renderer/components/+workloads-overview/overview-workload-status.tsx [hunk 1]
```diff
@@ -10,90 +10,105 @@ import capitalize from "lodash/capitalize";
...
-          },
-        }}
-      />
-    );
+    chartData.datasets.push({
+      data,
+      backgroundColor,
+      label: "Status",
+      tooltipLabels,
+    });
```
Comment (jansav) [inline src/renderer/components/+workloads-overview/overview-workload-status.tsx]: This part could be simplified a lot.

Perhaps something like this?

```typescript
const statusesToBeShown = Object.entries(workload.status.get()).filter(
  ([, value]) => value > 0
);

const theme = activeTheme.get();

const emptyDataSet = {
  data: [1],
  backgroundColor: [theme.colors.pieChartDefaultColor],
  label: "Empty",
};

const statusDataSet = {
  label: "Status",

  data: statusesToBeShown.map(([, value]) => value),

  backgroundColor: statusesToBeShown.map(([status]) => {
    // Typing needed here
    const color = statusBackgroundColorMapping[toLowercase(status)];

    return theme.colors[color];
  }),

  tooltipLabels: statusesToBeShown.map(
    ([status]): DatasetTooltipLabel => (percent) =>
      `${capitalize(status)}: ${percent}`
  ),
};

const chartData: Required<PieChartData> = {
  datasets: [statusesToBeShown.length > 0 ? statusDataSet : emptyDataSet],

  labels: statusesToBeShown.map(
    ([status, value]) => `${capitalize(status)}: ${value}`
  ),
};
```

File: src/renderer/components/chart/pie-chart.tsx [hunk 2]
```diff
@@ -68,18 +69,11 @@ const NonInjectedPieChart = observer(({
           const total = datasetData.reduce((acc, cur) => acc + cur, 0);
           const percent = Math.round((datasetData[tooltipItem.index] as number / total) * 100);
           const percentLabel = isNaN(percent) ? "N/A" : `${percent}%`;
-          const tooltipLabel = dataset.tooltipLabels?.[tooltipItem.index];
-          let tooltip = `${dataset.label}: ${percentLabel}`;
+          const computeTooltipLabel = dataset.tooltipLabels?.[tooltipItem.index];
```
Comment (jansav) [inline src/renderer/components/chart/pie-chart.tsx]: nit: `compute` gives wrong impression.

File: src/behaviours/cluster/workload-overview.test.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  describe("when navigating to workload overview", () => {
+    beforeEach(() => {
+      applicationBuilder.navigateWith(navigateToWorkloadsOverviewInjectable);
+    });
+
+    it("renders", () => {
+      expect(rendered.baseElement).toMatchSnapshot();
+    });
+
+    it("does not yet show workload overview", () => {
```
Comment (jansav) [inline src/behaviours/cluster/workload-overview.test.tsx]: Hmm? Why we are not showing the workload overview?

Comment (Nokel81) [reply src/behaviours/cluster/workload-overview.test.tsx]: Test name is wrong, the test itself checks for the opposite condition.

**Review by aleksfront — DISMISSED** [review]
LGTM. A note for testers: links are broken in Overview page, but it was fixed by https://github.com/lensapp/lens/pull/6002 and problem will disappear after merge or rebase.

**Review by aleksfront — APPROVED** [review]

**Comment by jim-docker** [comment]
I still don't think this should go in a patch release. The linked issue is targeted for 6.1, which seems appropriate given the low severity of the issue, and the large number of files changed here to fix it.

**Comment by Nokel81** [comment]
Moved to 6.1

**Review by jim-docker — APPROVED** [review]

**Pull request merged**